metacat-connector-s3/src/main/java/com/netflix/metacat/connector/s3/S3ConnectorDatabaseService.java

```diff
@@ -0,0 +1,150 @@
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * S3 Connector Database Service implementation.
+ * @author amajumdar
+ */
+@Transactional
+public class S3ConnectorDatabaseService implements ConnectorDatabaseService {
+    private final SourceDao sourceDao;
+    private final DatabaseDao databaseDao;
+    private final S3ConnectorInfoConverter infoConverter;
+    private final String catalogName;
+
+    /**
+     * Constructor.
+     * @param catalogName catalog name
+     * @param databaseDao database DAO impl
+     * @param sourceDao catalog/source DAO impl
+     * @param infoConverter Converter for the S3 resources
+     */
+    @Inject
+    public S3ConnectorDatabaseService(@Named("catalogName") final String catalogName, final DatabaseDao databaseDao,
+        final SourceDao sourceDao, final S3ConnectorInfoConverter infoConverter) {
+        this.databaseDao = databaseDao;
+        this.sourceDao = sourceDao;
+        this.infoConverter = infoConverter;
+        this.catalogName = catalogName;
+    }
+
+    @Override
+    public List<QualifiedName> listViewNames(@Nonnull final ConnectorContext context,
+        @Nonnull final QualifiedName databaseName) {
+        return null;
+    }
+
+    @Override
+    public void create(@Nonnull final ConnectorContext context, @Nonnull final DatabaseInfo databaseInfo) {
+        final String databaseName = databaseInfo.getName().getDatabaseName();
+        Preconditions.checkNotNull(databaseName, "Schema name is null");
+        if (databaseDao.getBySourceDatabaseName(catalogName, databaseName) != null) {
+            throw new DatabaseAlreadyExistsException(databaseInfo.getName());
+        }
+        final Database database = new Database();
+        database.setName(databaseName);
+        database.setSource(sourceDao.getByName(catalogName));
+        databaseDao.save(database);
+    }
+
+    @Override
+    public void update(@Nonnull final ConnectorContext context, @Nonnull final DatabaseInfo databaseInfo) {
+        // no op
+    }
+
+    @Override
+    public void delete(@Nonnull final ConnectorContext context, @Nonnull final QualifiedName name) {
+        final String databaseName = name.getDatabaseName();
+        Preconditions.checkNotNull(databaseName, "Schema name is null");
+        final Database database = databaseDao.getByName(databaseName);
+        if (database == null) {
+            throw new DatabaseNotFoundException(name);
+        } else if (database.getTables() != null && !database.getTables().isEmpty()) {
+            throw new ConnectorException("Database " + databaseName + " is not empty. One or more tables exist.", null);
+        }
+        databaseDao.delete(database);
+    }
+
+    @Override
+    public DatabaseInfo get(@Nonnull final ConnectorContext context, @Nonnull final QualifiedName name) {
+        final String databaseName = name.getDatabaseName();
+        Preconditions.checkNotNull(databaseName, "Schema name is null");
```

Comment:
Didn't we agree to name everything `database` shouldn't this say database not schema?